mysql-test/include/forbid_same_constraint_name_mdev18463.inc

```diff
@@ -0,0 +1,78 @@
+--source include/have_innodb.inc
+
+eval CREATE TABLE tpk
+(id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,name VARCHAR(100) NOT NULL)
+ENGINE=$engine;
+
+--echo #
+--echo # Same name `sid` for check and foreign key constraint while creating the table
+--echo #
+--error ER_DUP_CONSTRAINT_NAME
+eval CREATE TABLE tfk
+(id INT, c1 INT, c2 INT NOT NULL,
+ CONSTRAINT sid FOREIGN KEY (`c1`) REFERENCES tpk (`id`)
+ ON DELETE NO ACTION ON UPDATE NO ACTION, CONSTRAINT sid check (c2>15))
+ ENGINE=$engine;
+
+# Create table with unique foreign key
+eval CREATE TABLE tfk (c1 INT, c2 INT, CONSTRAINT sid FOREIGN KEY (c1)
+                       REFERENCES tpk (id)) ENGINE=$engine;
+SHOW CREATE TABLE tfk;
+
+--echo #
+--echo # Add check constraint with the same name `sid` as foreign key
+--echo #
+--error ER_DUP_CONSTRAINT_NAME
+ALTER TABLE tfk ADD CONSTRAINT Sid CHECK (c2>15);
+SHOW CREATE TABLE tfk;
+
+--echo #
+--echo # Add new foreign constraint with the same name `sid` as foreign key
+--echo #
+--error ER_DUP_CONSTRAINT_NAME
+ALTER TABLE tfk ADD CONSTRAINT Sid CHECK (c2>20);
+
+# This doesn't drop index KEY('sid')
+ALTER TABLE tfk DROP FOREIGN KEY sid;
+SHOW CREATE TABLE tfk;
+
+--echo #
+--echo # Drop the foreign key will not drop the index so adding new constraint
+--echo # will raise an error
+--echo #
+--error ER_DUP_CONSTRAINT_NAME
+ALTER TABLE tfk ADD CONSTRAINT Sid CHECK (c2>15);
```

Comment:
Is a non-unique index really a constraint? If `ADD CONSTRAINT…FOREIGN KEY` is able to add implicit `KEY` with the same name, isn’t that a proof that index names and constraint names are disjoint domains?

I might accept that `UNIQUE` index counts as a `CONSTRAINT` name, but what does the standard say?

Reply:
AFAIK the SQL standard doesn't mention non-unique index constraint.